internal/wasm/memory.go

```diff
@@ -284,3 +292,49 @@ func (m *MemoryInstance) writeUint64Le(offset uint32, v uint64) bool {
 	binary.LittleEndian.PutUint64(m.Buffer[offset:], v)
 	return true
 }
+
+// Wait suspends the caller until the offset is notified by a different agent.
+func (m *MemoryInstance) Wait(offset uint32, timeout int64) (bool, bool) {
+	m.Mux.Lock()
+
+	if m.waiters == nil {
+		m.waiters = make(map[uint32][]chan struct{})
+	}
+
+	if uint(len(m.waiters[offset])) == math.MaxUint32 {
```

Comment:
Why `math.MaxUint32` as the limit? I guess I need a comment.
Append will panic way before this on 32-bit, and 4 billion listeners seems way too much…?